src/Evolution/Systems/ScalarTensor/BoundaryConditions/ProductOfConditions.hpp

```diff
@@ -0,0 +1,373 @@
+#include "PointwiseFunctions/GeneralRelativity/Shift.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpatialMetric.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace ScalarTensor::BoundaryConditions {
+namespace detail {
+
+template <evolution::BoundaryConditions::Type GhBcType,
+          evolution::BoundaryConditions::Type ScalarBcType>
+struct UnionOfBcTypes {
+  static_assert(
+      (GhBcType == evolution::BoundaryConditions::Type::TimeDerivative) or
+          (ScalarBcType ==
+           evolution::BoundaryConditions::Type::TimeDerivative) or
+          (GhBcType ==
+           evolution::BoundaryConditions::Type::GhostAndTimeDerivative) or
+          (ScalarBcType ==
+           evolution::BoundaryConditions::Type::GhostAndTimeDerivative),
+      "TimeDerivative boundary conditions are not yet supported.");
+  static constexpr evolution::BoundaryConditions::Type bc_type =
+      evolution::BoundaryConditions::Type::Ghost;
+};
+
+template <>
+struct UnionOfBcTypes<evolution::BoundaryConditions::Type::Ghost,
+                      evolution::BoundaryConditions::Type::Ghost> {
+  static constexpr evolution::BoundaryConditions::Type bc_type =
+      evolution::BoundaryConditions::Type::Ghost;
+};
+
+template <evolution::BoundaryConditions::Type GhBcType>
+struct UnionOfBcTypes<
+    GhBcType, evolution::BoundaryConditions::Type::DemandOutgoingCharSpeeds> {
+  static_assert(
+      GhBcType == evolution::BoundaryConditions::Type::DemandOutgoingCharSpeeds,
+      "If either boundary condition in `ProductOfConditions` has "
+      "`Type::DemandOutgoingCharSpeeds`, both must have "
+      "`Type::DemandOutgoingCharSpeeds`");
+};
+
+template <evolution::BoundaryConditions::Type ScalarBcType>
+struct UnionOfBcTypes<
+    evolution::BoundaryConditions::Type::DemandOutgoingCharSpeeds,
+    ScalarBcType> {
+  static_assert(
+      ScalarBcType ==
+          evolution::BoundaryConditions::Type::DemandOutgoingCharSpeeds,
+      "If either boundary condition in `ProductOfConditions` has "
+      "`Type::DemandOutgoingCharSpeeds`, both must have "
+      "`Type::DemandOutgoingCharSpeeds`");
+};
+
+template <>
+struct UnionOfBcTypes<
+    evolution::BoundaryConditions::Type::DemandOutgoingCharSpeeds,
+    evolution::BoundaryConditions::Type::DemandOutgoingCharSpeeds> {
+  static constexpr evolution::BoundaryConditions::Type bc_type =
+      evolution::BoundaryConditions::Type::DemandOutgoingCharSpeeds;
+};
+
+}  // namespace detail
+
+/*!
+ * \brief Apply a boundary condition to the combined Generalized Harmonic (GH)
+ * and CurvedScalarWave (CSW) system using the boundary conditions defined
+ * separately for the GH and CSW systems.
+ */
+template <typename DerivedGhCondition, typename DerivedScalarCondition>
+class ProductOfConditions final : public BoundaryCondition {
+ public:
+  static constexpr size_t Dim = 3;
```

Comment:
`Dim` -> `dim` or just hard code to 3? Is there any reason to want to evolve this system in fewer dims?

Reply:
I actually wanted to avoid templating in other dimensions. But was asked to add DIm (or dim) for things in previous reviews.

Reply:
Hmm, okay, the code base generally just hard codes the dim if the system is only valid in one. You got advice not in line with how most of the code is done. Sorry about that!